src/librustc/ty/normalize_erasing_regions.rs

```diff
@@ -94,6 +94,10 @@ impl TypeFolder<'tcx> for NormalizeAfterErasingRegionsFolder<'tcx> {
     }
 
     fn fold_ty(&mut self, ty: Ty<'tcx>) -> Ty<'tcx> {
-        self.tcx.normalize_ty_after_erasing_regions(self.param_env.and(ty))
+        self.tcx.normalize_generic_arg_after_erasing_regions(self.param_env.and(GenericArg::from(ty))).expect_ty()
```

Comment:
You can write `ty.into()` instead of `GenericArg::from(ty)`.

Reply:
I actually favor being explicit here. In case you have a strong opinion on this I will ofc comply :laughing:

Reply:
We use `.into()` throughout the compiler for this. The type doesn't really matter, if we could make implicit coercion do the conversion, we would.